### DISCUSSION TITLE
My filter doesn't count correctly

### DISCUSSION THREAD

**Discussion by mcatalan-trioteca** [laravel/nova-issues#6067] [Q&A] [answered]
I have a custom filter with a date range and the interface doesn't count well when clearing the values.

We execute these lines of code when changing the value of the filter, but the number of the applied filters never decreases.

```
this.$store.commit(`${this.resourceName}/updateFilterState`, {
  filterClass: this.filterKey,
  value,
});
```

Does anyone have an idea what might be missing?

Thanks for all.

https://github.com/laravel/nova-issues/assets/107840986/02ff9d27-d30b-43f6-8776-09b82b1a6d15

**Comment by davidhemphill** [comment, marked as answer]
You need to ensure you're emitting the `change` event in your filter: `this.$emit('change')`

**Reply by davidhemphill** [reply]
Also, you need to implement the `handleFilterReset` in your component so it'll clear its value when the user clicks the "Reset Filter" button.
